### PR TITLE
Memoize and unit tests on screens mapStateToProps

### PR DESCRIPTION
**Pull request by VincentCATILLON**
## Detailed purpose of the PR

Add memoize and unit tests on all screens mapStateToProps

### Testing strategy

- [ ] I ensured my code is tested: unit tests
- [ ] I took care to mention breaking changes and extra libs if appropriate

### CODE REVIEW

**Comment by VincentCATILLON** [comment]
Missing things:
- Lesson resource type is not as expected (?)
- Memoize on Correction and Question to be done

File: jest.config.unit.js [hunk 1]
```diff
@@ -35,21 +35,21 @@ module.exports = {
       // @todo increase this value
       branches: 40,
       // @todo increase this value
-      functions: 42,
+      functions: 41,
       // @todo increase this value
       lines: 46,
       // @todo increase this value
-      statements: 47
+      statements: 46
```
Comment (VincentCATILLON) [inline jest.config.unit.js]: This value decreased because I moved some state extractor from containers to state extractor file, so the percentage decreased from 46.0x to 45.9x 🙄

File: jest.config.unit.js [hunk 2]
```diff
@@ -35,21 +35,21 @@ module.exports = {
       // @todo increase this value
       branches: 40,
       // @todo increase this value
-      functions: 42,
+      functions: 41,
```
Comment (VincentCATILLON) [inline jest.config.unit.js]: This value decreased because I moved some state extractor from containers to state extractor file, so the percentage decreased from 42.0x to 41.9x 🙄

File: jest.config.unit.js [hunk 3]
```diff
@@ -33,23 +33,23 @@ module.exports = {
     },
     'src/containers': {
       // @todo increase this value
-      branches: 40,
+      branches: 39,
```
Comment (VincentCATILLON) [inline jest.config.unit.js]: This value decreased because I lightened `question-slider` container, so the percentage decreased from 40.0x to 39.9x 🙄

File: jest.config.unit.js (hunk 1 above)
Comment (Fausto95) [reply jest.config.unit.js]: Not sure about this

File: src/components/question-choices.js [hunk 4]
```diff
@@ -138,13 +140,15 @@ class QuestionChoices extends React.PureComponent<Props> {
           </View>
         );
       case QUESTION_TYPE.SLIDER: {
-        if (!min || !max) {
+        if (min === undefined || max === undefined) {
```
Comment (Fausto95) [inline src/components/question-choices.js]: Why this change?

**Comment by Volubyl** [comment]
I'm really understand the @Fausto95  point : the coverage is very pretty low ...

File: src/components/question-choices.js (hunk 4 above)
Comment (VincentCATILLON) [reply src/components/question-choices.js]: Because the unit data was redundant, `Edge` type was useless so, and default value from `mapStateToProps` was an object containing properties with `"undefined undefined"` instead of undefined props, as simple as it should.
